Formula/a/asdf.rb

```diff
@@ -16,38 +16,17 @@ class Asdf < Formula
     sha256 cellar: :any_skip_relocation, all: "a94dd362ce5c3a818f4fa56607eac3c616a3c1191e9f41480093bda2dc308af4"
   end
 
-  depends_on "autoconf"
-  depends_on "automake"
-  depends_on "coreutils"
-  depends_on "libtool"
-  depends_on "libyaml"
+  depends_on "go" => :build
+
   depends_on "openssl@3"
```

Comment:
Is it really depending on openssl library? That is usually not needed for Go programs and ssl is not being mentioned upstream either: https://github.com/search?q=repo%3Aasdf-vm%2Fasdf%20ssl&type=code

If it is really needed I would expect it to be mentioned in some build & test workflows
https://github.com/asdf-vm/asdf/tree/master/.github/workflows

Reply:
I thought it was needed by `go-git`, but it looks like go-git doesn't actually require it to be present during compilation. PR has been updated.

Reply:
I think a lot of dependencies were originally for plugins (Python, Ruby, etc) which would use them similar to `pyenv`/`ruby-build`/etc.

Whether these should be explicit dependencies though was always a question:
* One on side, we could keep dependencies minimal and expect users to install anything they need
* On other side was the idea of making it easier for users to run commands like `asdf plugin add ruby && asdf install ruby ...` "out-of-the-box".

Reply:
Yes, I believe that was the intent. And early on there were less than a dozen plugins, so having dependencies that covered most of them was easy. Now there are hundreds (appears to be approaching 1k) so installing dependencies even for a small fraction of them is out of the question. I think we should keep the dependencies here as minimal as possible.

Reply:
Thanks! Indeed, it might be best to leave such dependancies to plugins, not the base asdf.